mod_mirrorbrain/mod_mirrorbrain.c

```diff
@@ -1892,10 +1892,14 @@ static int mb_handler(request_rec *r)
 
     realfile = ptr;
     debugLog(r, cfg, "Canonicalized file on disk: %s", realfile);
-
-    /* the leading directory needs to be stripped from the file path */
-    /* a directory from Apache always ends in '/'; a result from realpath() doesn't */
-    filename = realfile + strlen(mirror_base) + 1;
+    if (strncmp(ptr, mirror_base, strlen(mirror_base))) {
+        debugLog(r, cfg, "File is not on mirror base -> using original name");
```

Comment:
how do we get the relative filename in the symlink case then?

Reply:
In my understanding we don't need that. We just treat filename as regular file and check if that exists in DB / on mirrors. Any better ideas?

Reply:
It will still be needed. The DB only has relative path names. And we have some trees on mirrorbrain servers which are constructed with lots of symlinks.

Reply:
so I checked the DB from download.o.o

we have the symlinked and the non symlinked paths in the DB right now. if we keep that behavior, it costs us some more disk space for the DB but saves us the whole making path canonical and stripping the path prefix. we just look up ```request->uri``` in the DB.

https://ci.apache.org/projects/httpd/trunk/doxygen/structrequest__rec.html#a200869c456b0bd046a32942f9ff51af4